core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java

```diff
@@ -232,4 +242,61 @@ public static boolean isAfterImageColumn(String columnName, TableMetadata tableM
     }
     return !isBeforeImageColumn(columnName, tableMetadata);
   }
+
+  static Put createPutForInsert(Insert insert) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(insert.forNamespace().orElse(null))
+            .table(insert.forTable().orElse(null))
+            .partitionKey(insert.getPartitionKey());
+    insert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    insert.getColumns().values().forEach(buildable::value);
+    buildable.enableInsertMode();
+    return buildable.build();
+  }
+
+  static Put createPutForUpsert(Upsert upsert) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(upsert.forNamespace().orElse(null))
+            .table(upsert.forTable().orElse(null))
+            .partitionKey(upsert.getPartitionKey());
+    upsert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    upsert.getColumns().values().forEach(buildable::value);
+    buildable.enableImplicitPreRead();
+    return buildable.build();
+  }
+
+  static Put createPutForUpdate(Update update) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(update.forNamespace().orElse(null))
+            .table(update.forTable().orElse(null))
+            .partitionKey(update.getPartitionKey());
+    update.getClusteringKey().ifPresent(buildable::clusteringKey);
+    update.getColumns().values().forEach(buildable::value);
+    if (update.getCondition().isPresent()) {
+      if (update.getCondition().get() instanceof UpdateIf) {
+        update
+            .getCondition()
+            .ifPresent(c -> buildable.condition(ConditionBuilder.putIf(c.getExpressions())));
+      } else {
+        assert update.getCondition().get() instanceof UpdateIfExists;
+        buildable.condition(ConditionBuilder.putIfExists());
+      }
+    } else {
+      buildable.condition(ConditionBuilder.putIfExists());
```

Comment:
This condition is implicitly added, but `ConsensusCommit.update()` checks if the original `Update` has any condition not a converted `Put` when `UnsatisfiedConditionException` is thrown from `CrudHandler.put()`.

So, this 3 is achieved, right?

1. When `UpdateIf` is specified for Update:
  i. If the target record doesn't exist ===> `UnsatisfiedConditionException`
  ii. If the condition isn't satisfied ===> `UnsatisfiedConditionException`
2. When `UpdateIfExists` is specified for Update:
  i. If the target record doesn't exist (and the condition isn't satisfied) ===> `UnsatisfiedConditionException`
3. When no condition is specified ===> No exceptions are thrown in any case

Reply:
Yes, this is for 3. When no condition is specified, the Update operation is executed only when the target record exists, and if the target record doesn't exist, we do nothing.